Allow doctrine/event-manager 2

franmomu opened the pull request:
<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | feature
| BC Break     | no
| Fixed issues | <!-- use #NUM format to reference an issue -->

#### Summary

I haven't found any calls to `EventManager::getListeners()` method.

franmomu committed:
Allow doctrine/event-manager 2

malarzm commented:
@franmomu any opinion on hasListeners usage? I was thinking we could get rid of it https://github.com/doctrine/mongodb-odm/issues/2478

franmomu commented:
> hasListeners

I didn't see the issue open 🤦 yeah I'll take a look at those cases

franmomu committed:
Remove unnecessary check

EventManager::dispatchEvent already checks if there are listeners for the event.

franmomu commented:
There is this one left:

https://github.com/doctrine/mongodb-odm/blob/f5c16795858ab537922ead80eb122927f6d78686/lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadataFactory.php#L90-L101

which maybe it's easier to read with the check first, but we could change it too.

franmomu commented on lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadataFactory.php:
```diff
@@ -202,12 +202,10 @@ protected function doLoadMetadata($class, $parent, $rootEntityFound, array $nonS
+        );
 
         // phpcs:ignore SlevomatCodingStandard.ControlStructures.EarlyExit.EarlyExitNotUsed
         if ($class->isChangeTrackingNotify()) {
```
This looks like it should have been called always not only when there are no event listeners for `Events::loadClassMetadata`.

malarzm replied:
Good catch 👍

malarzm commented:
Re `onNotFoundMetadata` method: yeah, this seems more readbale this way :) Let's keep it as is.

malarzm commented:
Thanks @franmomu!
